pkg/function/backup_data_using_kopia_server.go

```diff
@@ -0,0 +1,260 @@
+		return nil, errors.Wrap(err, "Failed to fetch Kopia API Server Certificate Secret Data from Certificate")
+	}
+
+	username := tp.RepositoryServer.Username
+	hostname, userAccessPassphrase, err := getHostNameAndUserPassPhraseFromRepoServer(userPassphrase)
+	if err != nil {
+		return nil, errors.Wrap(err, "Failed to fetch Hostname/User Passphrase from Secret")
+	}
+
+	cli, err := kube.NewClient()
+	if err != nil {
+		return nil, errors.Wrap(err, "Failed to create Kubernetes client")
+	}
+
+	ctx = field.Context(ctx, consts.PodNameKey, pod)
+	ctx = field.Context(ctx, consts.ContainerNameKey, container)
+
+	serverAddress, err := getRepositoryServerAddress(cli, *tp.RepositoryServer)
+	if err != nil {
+		return nil, errors.Wrap(err, "Failed to get the Kopia Repository Server Address")
+	}
+
+	snapInfo, err := backupDataUsingKopiaServer(
+		cli,
+		container,
+		hostname,
+		includePath,
+		namespace,
+		pod,
+		serverAddress,
+		fingerprint,
+		username,
+		userAccessPassphrase,
+		tags,
+	)
+	if err != nil {
+		return nil, errors.Wrap(err, "Failed to backup data using Kopia Repository Server")
+	}
+
+	var logSize, phySize int64
+	if snapInfo.Stats != nil {
+		stats := snapInfo.Stats
+		logSize = stats.SizeHashedB + stats.SizeCachedB
+		phySize = stats.SizeUploadedB
+	}
+
+	output := map[string]any{
+		BackupDataOutputBackupID:           snapInfo.SnapshotID,
+		BackupDataOutputBackupSize:         humanize.Bytes(uint64(logSize)),
+		BackupDataOutputBackupPhysicalSize: humanize.Bytes(uint64(phySize)),
+	}
+	return output, nil
+}
+
+func backupDataUsingKopiaServer(
+	cli kubernetes.Interface,
+	container,
+	hostname,
+	includePath,
+	namespace,
+	pod,
+	serverAddress,
+	fingerprint,
+	username,
+	userPassphrase string,
+	tags []string,
+) (info *kopiacmd.SnapshotCreateInfo, err error) {
+	contentCacheMB, metadataCacheMB := kopiacmd.GetCacheSizeSettingsForSnapshot()
+	configFile, logDirectory := kankopia.GetCustomConfigFileAndLogDirectory(hostname)
+
+	cmd := kopiacmd.RepositoryConnectServerCommand(kopiacmd.RepositoryServerCommandArgs{
+		UserPassword:    userPassphrase,
+		ConfigFilePath:  configFile,
+		LogDirectory:    logDirectory,
+		CacheDirectory:  kopiacmd.DefaultCacheDirectory,
+		Hostname:        hostname,
+		ServerURL:       serverAddress,
+		Fingerprint:     fingerprint,
+		Username:        username,
+		ContentCacheMB:  contentCacheMB,
+		MetadataCacheMB: metadataCacheMB,
+	})
+
+	stdout, stderr, err := kube.Exec(cli, namespace, pod, container, cmd, nil)
+	format.Log(pod, container, stdout)
+	format.Log(pod, container, stderr)
+	if err != nil {
+		return nil, errors.Wrap(err, "Failed to connect to Kopia Repository Server")
+	}
+
+	cmd = kopiacmd.SnapshotCreate(
+		kopiacmd.SnapshotCreateCommandArgs{
+			PathToBackup: includePath,
+			CommandArgs: &kopiacmd.CommandArgs{
+				RepoPassword:   "",
+				ConfigFilePath: configFile,
+				LogDirectory:   logDirectory,
+			},
+			Tags:                   tags,
+			ProgressUpdateInterval: 0,
```

Comment:
Don't we need to set the `Parallelism:` field?

Reply:
Will have to look into it. I think we take the Parallelism's value using ENV Variables. But I'm not sure here in Kanister what's the Key of the Environmental variable. Or are we setting it at all. I think @ankitjain235 might have an idea.